File: src/MultiAgentApp/Configuration/AIOptions.cs
namespace MultiAgentApp.Configuration;

/// <summary>
/// Top-level AI provider options.
/// Toggle <see cref="UseFoundryLocal"/> to switch between Microsoft Foundry Local
/// (running on the developer's machine) and Azure AI Foundry (cloud).
/// </summary>
public sealed class AIOptions
{
    public const string SectionName = "AI";

    /// <summary>
    /// When <c>true</c>, the app connects to a locally-running Microsoft Foundry Local instance.
    /// Set to <c>false</c> to use Azure AI Foundry once a cloud environment is available.
    /// </summary>
    public bool UseFoundryLocal { get; set; } = true;

    /// <summary>Connection settings for Microsoft Foundry Local (local inference).</summary>
    public FoundryLocalOptions FoundryLocal { get; set; } = new();

    /// <summary>Connection settings for Azure AI Foundry (cloud). Used when <see cref="UseFoundryLocal"/> is <c>false</c>.</summary>
    public AzureAIFoundryOptions AzureAIFoundry { get; set; } = new();
}

/// <summary>
/// Connection settings for Microsoft Foundry Local.
/// Foundry Local exposes an OpenAI-compatible REST API, typically at http://localhost:5272/v1.
/// </summary>
public sealed class FoundryLocalOptions
{
    /// <summary>Base URL of the Foundry Local OpenAI-compatible endpoint.</summary>
    public string Endpoint { get; set; } = "http://localhost:5272/v1";

    /// <summary>
    /// Name of the model loaded in Foundry Local.
    /// List available models with: <c>foundry model list</c>.
    /// </summary>
    public string ModelId { get; set; } = "phi-4-mini-reasoning";

    /// <summary>
    /// Placeholder API key value (Foundry Local does not require authentication,
    /// but the OpenAI client library requires a non-empty value).
    /// </summary>
    public string ApiKey { get; set; } = "foundry-local";
}

/// <summary>
/// Connection settings for Azure AI Foundry.
/// Populated from environment variables or Azure Key Vault once a cloud environment is available.
/// </summary>
public sealed class AzureAIFoundryOptions
{
    /// <summary>Azure OpenAI / AI Foundry endpoint URL.</summary>
    public string Endpoint { get; set; } = string.Empty;

    /// <summary>Chat completion deployment name in Azure AI Foundry.</summary>
    public string DeploymentName { get; set; } = string.Empty;

    /// <summary>
    /// Azure OpenAI API key.
    /// In production, retrieve this from Azure Key Vault or Managed Identity rather than storing it here.
    /// </summary>
    public string ApiKey { get; set; } = string.Empty;
}

File: src/multi_agent_app/config/ai_options.py
"""Configuration dataclasses for AI provider settings."""

from __future__ import annotations
from dataclasses import dataclass, field


@dataclass
class FoundryLocalOptions:
    """Connection settings for Microsoft Foundry Local (local inference).

    Foundry Local exposes an OpenAI-compatible REST API, typically at
    http://localhost:5272/v1.  No authentication is required; ``api_key`` is a
    placeholder satisfying the OpenAI client library validation.
    """

    endpoint: str = "http://localhost:5272/v1"
    model_id: str = "phi-4-mini-reasoning"
    api_key: str = "foundry-local"

    @classmethod
    def from_dict(cls, d: dict) -> FoundryLocalOptions:
        return cls(
            endpoint=d.get("Endpoint", "http://localhost:5272/v1"),
            model_id=d.get("ModelId", "phi-4-mini-reasoning"),
            api_key=d.get("ApiKey", "foundry-local"),
        )


@dataclass
class MicrosoftFoundryOptions:
    """Connection settings for Microsoft Foundry (cloud).

    Populate from environment variables or Azure Key Vault once a cloud
    environment is available.  Leave ``api_key`` empty to use
    ``DefaultAzureCredential`` (Managed Identity / Azure CLI / env vars).
    """

    endpoint: str = ""
    deployment_name: str = ""
    project_name: str = ""
    api_key: str = ""

    @classmethod
    def from_dict(cls, d: dict) -> MicrosoftFoundryOptions:
        return cls(
            endpoint=d.get("Endpoint", ""),
            deployment_name=d.get("DeploymentName", ""),
            project_name=d.get("ProjectName", ""),
            api_key=d.get("ApiKey", ""),
        )


@dataclass
class AIOptions:
    """Top-level AI provider options.

    Toggle ``use_foundry_local`` to switch between Microsoft Foundry Local
    (running on the developer's machine) and Microsoft Foundry (cloud).
    """

    SECTION_NAME: str = field(default="AI", init=False, repr=False, compare=False)

    use_foundry_local: bool = True
    foundry_local: FoundryLocalOptions = field(default_factory=FoundryLocalOptions)
    microsoft_foundry: MicrosoftFoundryOptions = field(
        default_factory=MicrosoftFoundryOptions
    )

    @classmethod
    def from_dict(cls, d: dict) -> AIOptions:
        return cls(
            use_foundry_local=d.get("UseFoundryLocal", True),
            foundry_local=FoundryLocalOptions.from_dict(d.get("FoundryLocal", {})),
            microsoft_foundry=MicrosoftFoundryOptions.from_dict(
                d.get("MicrosoftFoundry", {})
            ),
        )
